api/spec/scanner-adapter-openapi-v1.2.yaml

```diff
@@ -0,0 +1,551 @@
+      type: object
+      properties:
+        type:
+          type: string
+          enum:
+          - vulnerability
+          - sbom
+          description: |
+            The type of the capability, for example, 'vulnerability' represents analyzing the artifact then producing the vulnerabilities report,
+            'sbom' represents generating the corresponding sbom for the artifact which be scanned. In order to the backward and forward compatible,
+            the field is optional, we think it's a original 'vulnerability' scan if no such field.
+        consumes_mime_types:
+          type: array
+          items:
+            type: string
+          description: |
+            The set of MIME types of the artifacts supported by the scanner to produce the reports specified in the "produces_mime_types". A given
+            mime type should only be present in one capability item.
+          example:
+            - "application/vnd.oci.image.manifest.v1+json"
+            - "application/vnd.docker.distribution.manifest.v2+json"
+        produces_mime_types:
+          type: array
+          items:
+            type: string
+          description: |
+            The set of MIME types of reports generated by the scanner for the consumes_mime_types of the same capability record.
+          example:
+            - "application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0"
+    ScanRequest:
+      required:
+        - registry
+        - artifact
+      type: object
+      properties:
+        registry:
+          $ref: '#/components/schemas/Registry'
+        artifact:
+          $ref: '#/components/schemas/Artifact'
+        enabled_capabilities:
+          type: array
+          description: Enable which capabilities supported by scanner, for backward compatibility, without this field scanner can be considered to enable all capabilities by default.
+          items:
+            type: object
+            properties:
+              type:
+                type: string
+                enum:
+                - vulnerability
+                - sbom
+                description: The type of the scan capability.
+                example: sbom
+              produces_mime_types:
+                type: array
+                items:
+                  type: string
+                description: |
+                  The set of MIME types of reports generated by the scanner for the consumes_mime_types of the same capability record, it is a subset or fullset of the
+                  produces_mime_types of the capability returned by the metadata API, used for client to fine grained control of the expected report type. It's a optional
+                  field, only applied when client needs to customize it, otherwise the scanner can think it's a fullset as before behavior if without this field.
+                example:
+                  - "application/vnd.security.sbom.report+json; version=1.0"
+              parameters:
```

Comment:
It's optional, right? We may want to define which parameters are required if `enabled_capabilities` is passed.

- type: required
- produces_mime_types: required?
- parameters: optional?

Reply:
only `type` is required, `produces_mime_types` is not required because by default it's returned by the `/metadata` API.

Reply:
So, should we add `required` as below?

```
required:
  - type
```

Reply:
yep, updated.